### PR TITLE
Javascript javascript3 week3

### PR DESCRIPTION
**Pull request by viedamayanti**

### CODE REVIEW

File: javascript/javascript3/week3/CirclePaint/index.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
```
Comment (Yuliia-Balandiuk) [inline javascript/javascript3/week3/CirclePaint/index.html:1]: Hi Vie, good job!
Here are some comments from me

File: javascript/javascript3/week3/CirclePaint/script.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    g = (255 * Math.random()) | 0,
+    b = (255 * Math.random()) | 0;
+  return 'rgb(' + r + ',' + g + ',' + b + ')';
+}
+randomColor();
+
+function randomPosition() {
+  random_x = Math.random() * windowHeight;
+  random_y = Math.random() * windowWidth;
+  random_r = Math.random() * 100;
```
Comment (Yuliia-Balandiuk) [inline javascript/javascript3/week3/CirclePaint/script.js:42]: If you a little change code in lines 40 and 41, circles will be drawn across the entire width of the screen.

random_x = Math.random() * windowWidth;
random_y = Math.random() * windowHeight;

File: javascript/javascript3/week3/GitFetch/index.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Document</title>
```
Comment (Yuliia-Balandiuk) [inline javascript/javascript3/week3/GitFetch/index.html:7]: You can use a batter title)

File: javascript/javascript3/week3/GitFetch/script.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  
+  const getAllRepos = async function () {
+    let data = await Promise.all([gitUser1, gitUser2, gitUser3]);
+    for (let item of data) {
+      const getName = item.items[0].full_name;
+      const getOwner = item.items[0].owner.login;
+      const getRepo = item.items[0].commits_url;
+      console.log(getName);
+      console.log(getOwner);
+      console.log(getRepo);
```
Comment (Yuliia-Balandiuk) [inline javascript/javascript3/week3/GitFetch/script.js:20]: It is better to delete unnecessary logs

File: javascript/javascript3/week3/GitFetch/script.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  
+      const headerStr = `These are github collection for`;
```
Comment (Yuliia-Balandiuk) [inline javascript/javascript3/week3/GitFetch/script.js:22]: You can also add the owner of the repo for better understanding. 
const headerStr = `These are github collection for ${getOwner}`;

File: javascript/javascript3/week3/ShoppingCart/index.html [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+          <button class="addBtn">Add</button>
+        </div>
+      </section>
+      <section class="products">
+        <ul id="ulList"></ul>
+      </section>
+      <!--<section class="cart">
+        <h2>Your shopping cart</h2>
+        <div class="total"></div>
+      </section> -->
```
Comment (Yuliia-Balandiuk) [inline javascript/javascript3/week3/ShoppingCart/index.html:49]: it is better to remove unnecessary code

File: javascript/javascript3/week3/ShoppingCart/style.css [hunk 7]
```diff
@@ -0,0 +1,73 @@
+* {
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+}
```
Comment (Yuliia-Balandiuk) [inline javascript/javascript3/week3/ShoppingCart/style.css:5]: Some reviewers say : 
Try also to practice putting properties in alphabetical order, like
* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

File: javascript/javascript3/week3/ShoppingCart/style.css [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+.searchBtn,
+.addBtn {
+  padding: 7px 12px;
+  background-color: #ce2129;
+  color: white;
+  border-radius: 6px;
+  font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
+  border: #ce2129 2px solid;
+}
```
Comment (Yuliia-Balandiuk) [inline javascript/javascript3/week3/ShoppingCart/style.css:45]: A good practice is to add cursor: pointer; to the btn;

File: javascript/javascript3/week3/ShoppingCart/script.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    this.name = name;
+    this.price = price;
+  }
+  convertToCurrency(currency) {
+    // Implement functionality here
+    const urlCurrency = `https://v6.exchangerate-api.com/v6/65e9169f45d75e71ef7414dd/latest/${currency}`;
+    return fetch(urlCurrency)
+      .then((response) => response.json())
+      .then((data) => {
+        const getConvert = data.conversion_rates[currency];
```
Comment (Yuliia-Balandiuk) [inline javascript/javascript3/week3/ShoppingCart/script.js:13]: Usually, we use "get" with functions. You can you for example convertRate

File: javascript/javascript3/week3/ShoppingCart/script.js [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+        //const convert = getConvert * this.price;
```
Comment (Yuliia-Balandiuk) [inline javascript/javascript3/week3/ShoppingCart/script.js:14]: try to remove the commented code

File: javascript/javascript3/week3/ShoppingCart/script.js [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+console.log(shoppingCart);
+const flatscreen = new Product('flat-screen', 5000);
+const plate = new Product('Plate', 300);
+const whitePlate = new Product('white plate', 70);
+shoppingCart.addProduct(flatscreen);
+shoppingCart.addProduct(plate);
+shoppingCart.addProduct(whitePlate);
+shoppingCart.renderProducts(whitePlate);
+shoppingCart.getUser(1);
+console.log(flatscreen.convertToCurrency('USD'));
```
Comment (Yuliia-Balandiuk) [inline javascript/javascript3/week3/ShoppingCart/script.js:86]: it doesn't work.  Can you check?

File: javascript/javascript3/week3/ShoppingCart/script.js [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+  convertToCurrency(currency) {
+    // Implement functionality here
+    const urlCurrency = `https://v6.exchangerate-api.com/v6/65e9169f45d75e71ef7414dd/latest/${currency}`;
+    return fetch(urlCurrency)
+      .then((response) => response.json())
+      .then((data) => {
+        const getConvert = data.conversion_rates[currency];
+        //const convert = getConvert * this.price;
+        console.log(getConvert);
+      });
```
Comment (semih1239) [inline javascript/javascript3/week3/ShoppingCart/script.js:16]: I think you should get all currency of one unit on the currency. you can use DKK here; we have to log the new prices with the currency rate. Also, you can easily call the function this way without having to console.log it. => `flatscreen.convertToCurrency('USD')`
```suggestion
  convertToCurrency(currency) {
    // Implement functionality here
    const urlCurrency = `https://v6.exchangerate-api.com/v6/65e9169f45d75e71ef7414dd/latest/DKK`;
    return fetch(urlCurrency)
      .then((response) => response.json())
      .then((data) => {
        const convertRate = data.conversion_rates[currency];
        console.log(this.price * convertRate )
      });
```

**Review by semih1239 — COMMENTED** [review]
Good job 💪 I wrote some comments. You can check them below.

File: javascript/javascript3/week3/CirclePaint/script.js (hunk 2 above)
Comment (viedamayanti) [reply javascript/javascript3/week3/CirclePaint/script.js:42]: Thank you so much Julia, it works fine now, its fill to the entire screen

File: javascript/javascript3/week3/ShoppingCart/script.js (hunk 11 above)
Comment (viedamayanti) [reply javascript/javascript3/week3/ShoppingCart/script.js:86]: No it does not work, but I already edit it

**Comment by viedamayanti** [comment]
Thank you so much for all the feedback, I learn a lot and I will implement all the feedback

**Pull request merged**